quay-pruner/build/prune-quay.py

```diff
@@ -66,41 +66,56 @@ def parse_quay_tag(tag):
     return ref, short_sha1, el, arch
 
 
+shaman_data = None
```

Comment:
Would you consider using something like [lru_cache](https://docs.python.org/3/library/functools.html#functools.lru_cache)?

Reply:
I added functools.cache().  It's a bit cleaner and gives me the opportunity to separate the caching and the searching.